openqaoa/optimizers/training_vqa.py

```diff
@@ -212,13 +218,15 @@ def optimize_this(self, x):
         if self.save_to_csv:
             save_parameter('param_log', deepcopy(x))
         
-        callback_cost = self.vqa.expectation(self.variational_params)
-        
+        callback_cost = self.vqa.expectation(self.variational_params)        
         log_dict.update({'cost': callback_cost})
+
         current_eval = self.log.func_evals.best[0]
         current_eval += 1
         log_dict.update({'func_evals': current_eval})
-        
+
+        log_dict.update({'eval_number': (current_eval - self.log.jac_func_evals.best[0] - self.log.qfim_func_evals.best[0])})  #this one will say which evaluation is the optimized one
```

Comment:
do we also need to subtract `self.log.hess_func_evals` or something similar? (For hessian evaluations, if specified)

Reply:
Maybe we should have more tests? Indeed I think Vishal is right, and the fact that the tests did not pick it up it means that the test can be improved

Reply:
I don't think `self.log.hess_func_evals` exists. When a hessian is called we update `self.log.jac_func_evals` too.

Reply:
I think the value is fine, but I will improve the tests, using different optimization methods.

Reply:
Great, thanks for the clarification!